packages/next/src/server-build.ts

```diff
@@ -945,6 +945,26 @@ export async function serverBuild({
             continue: true,
             ...(isOverride ? { override: true } : {}),
           },
+          // handle trailing slash
+          {
+            src: path.join('^/', entryDirectory, '((?!_next/).*)/$'),
+            has: [
+              {
+                type: 'header',
+                key: 'x-nextjs-data',
+              },
+            ],
+            dest: `${path.join(
+              '/',
+              entryDirectory,
+              '/_next/data/',
+              buildId,
+              '/$1.json'
+            )}`,
+            continue: true,
+            ...(isOverride ? { override: true } : {}),
+          },
+          // handle non-trailing slash
           {
             src: path.join('^/', entryDirectory, '((?!_next/).*)$'),
```

Comment:
Could we instead change this one to `'((?!_next/).*)/?$'`

I'd love to avoid adding another has route

Reply:
That doesn't exclude the trailing slash from the capture group correctly x-ref: https://regexr.com/6ogac

I initially tried just adding that as well.

Reply:
How about https://regexr.com/6ogao

Reply:
Nice, yeah looks like that works. Updated in https://github.com/vercel/vercel/pull/8015/commits/e5638fc1feeb72e829e74cae98dddf23b63fd632